catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -518,77 +498,44 @@ private void deleteListOfMetacards(
+        LOGGER.info("Failed to get list of Solr documents by ID for delete.", e);
+        throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
       }
-
+    } else {
       try {
-        deletedMetacards.add(client.createMetacard(doc));
-      } catch (MetacardCreationException e) {
-        LOGGER.info("Metacard creation exception creating metacards during delete", e);
+        return client.query(client.getIdentifierQuery(fieldName, identifierPaged));
+      } catch (UnsupportedQueryException e) {
+        LOGGER.info("Failed to get list of Solr documents for delete.", e);
         throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
       }
     }
   }
 
-  private void addPendingNrtDeletedMetacards(
-      List<Metacard> deletedMetacards, List<? extends Serializable> identifiers) {
-    if (deletedMetacards.size() < identifiers.size()
-        && pendingNrtIndex.getAllPresent(identifiers).size() > 0) {
-      Map<String, Metacard> matches = pendingNrtIndex.getAllPresent(identifiers);
-
-      deletedMetacards.forEach(m -> matches.remove(m.getId()));
-      deletedMetacards.addAll(matches.values());
-    }
-  }
-
-  private Metacard copyMetacard(Metacard original) {
-    MetacardImpl copy = new MetacardImpl(original, original.getMetacardType());
-    copy.setSourceId(original.getSourceId());
-    return copy;
-  }
-
-  private SolrDocumentList getSolrDocumentList(
-      List<? extends Serializable> identifierPaged, String fieldName) throws IngestException {
-    SolrQuery query = new SolrQuery(client.getIdentifierQuery(fieldName, identifierPaged));
-    query.setRows(identifierPaged.size());
-
-    QueryResponse solrResponse;
-    try {
-      solrResponse = solr.query(query, METHOD.POST);
-    } catch (SolrServerException | SolrException | IOException e) {
-      LOGGER.info("Failed to get list of Solr documents for delete.", e);
-      throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
-    }
-    return solrResponse.getResults();
-  }
-
-  private String getQuery(String attributeName, List<String> ids) throws IngestException {
-
-    StringBuilder queryBuilder = new StringBuilder();
+  private String getQuery(String attributeName, Set<String> ids) throws IngestException {
 
     List<String> mappedNames = resolver.getAnonymousField(attributeName);
 
     if (mappedNames.isEmpty()) {
       throw new IngestException("Could not resolve attribute name [" + attributeName + "]");
     }
 
-    for (int i = 0; i < ids.size(); i++) {
-
-      String id = ids.get(i);
-
-      if (i > 0) {
-        queryBuilder.append(" OR ");
-      }
-
-      queryBuilder.append(mappedNames.get(0)).append(":").append(QUOTE).append(id).append(QUOTE);
-    }
-
-    String query = queryBuilder.toString();
+    String query =
+        "{!terms cache=false separator=\" OR \" f="
+            + mappedNames.get(0)
+            + "}"
+            + String.join(" OR ", ids);
```

Comment:
LOL why " OR "?  Why not just a space?

Reply:
It is possible to override the format of IDs in the system.  I could not guarantee most common separators would not be used.  Since `" OR "` was already working I figured I would start with this for now.